uberfire-extensions/uberfire-metadata/uberfire-metadata-backends/uberfire-metadata-backend-infinispan/src/main/java/org/uberfire/ext/metadata/backend/infinispan/provider/InfinispanContext.java

```diff
@@ -0,0 +1,218 @@
+        });
+
+        SerializationContext serializationContext = ProtoStreamMarshaller.getSerializationContext(cacheManager);
+
+        addProtobufClass(serializationContext,
+                         SCHEMA_PROTO,
+                         Schema.class);
+    }
+
+    private void addProtobufClass(SerializationContext serializationContext,
+                                  String protoName,
+                                  Class<?> clazz) {
+        try {
+            ProtoSchemaBuilder protoSchemaBuilder = new ProtoSchemaBuilder();
+            protoSchemaBuilder.fileName(protoName);
+            protoSchemaBuilder.addClass(clazz);
+            String schemaString = protoSchemaBuilder.build(serializationContext);
+            this.getProtobufCache().put(protoName,
+                                        schemaString);
+        } catch (IOException e) {
+            throw new InfinispanException("Can't add protobuf class <" + protoName + "> to cache",
+                                          e);
+        }
+    }
+
+    private RemoteCache<String, String> getProtobufCache() {
+        return this.cacheManager.getCache(ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME);
+    }
+
+    public RemoteCache<String, KObject> getCache(String index) {
+        String i = AttributesUtil.toProtobufFormat(index).toLowerCase();
+        if (!this.getIndices().contains(i)) {
+            cacheManager
+                    .administration()
+                    .createCache(i,
+                                 this.infinispanConfiguration.getIndexedConfiguration(i));
+        }
+        return this.cacheManager.getCache(i);
+    }
+
+    public List<String> getTypes(String index) {
+        return this.getSchema(index)
+                .map(schema -> schema.getMessages().stream().map(x -> x.getName()).collect(toList()))
+                .orElse(Collections.emptyList());
+    }
+
+    public void addProtobufSchema(String clusterId,
+                                  Schema schema) {
+
+        try {
+            String protoTypeName = AttributesUtil.toProtobufFormat(clusterId);
+            RemoteCache<String, String> metadataCache = getProtobufCache();
+            String proto = this.schemaGenerator.generate(schema);
+
+            marshaller.registerSchema(protoTypeName,
+                                      proto,
+                                      KObject.class);
+            metadataCache.put(protoTypeName + PROTO_EXTENSION,
+                              proto);
+        } catch (IOException e) {
+            throw new InfinispanException("Can't add protobuf schema <" + schema.getName() + "> to cache",
+                                          e);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (this.cacheManager.isStarted()) {
+            this.cacheManager.stop();
+        }
+    }
+
+    public List<String> getIndices() {
+        return this.cacheManager.getCacheNames()
+                .stream()
```

Comment:
Why stream and then collect to list when you can just `return new ArrayList<>(cacheManager.getCacheNames());`

Reply:
I think I was doing more things and I forgot to change the back